pkg/webhook/conversion/chain.go

```diff
@@ -0,0 +1,251 @@
+package conversion
+
+import (
+	"strings"
+
+	"github.com/flant/shell-operator/pkg/utils/string_helper"
+)
+
+type ChainStorage map[string]*Chain
```

Comment:
It is better to make ChainStorage a structure with an internal field for the map to make it more flexible.
For example, with the current approach, we can't add a simple lock to protect Chains.

Reply:
Right now lock is not required for ChainStorage — it is filled once at startup. But it should be a good idea to add a lock for maps in Chain structure.

Reply:
A lock is just an example. We can not add new properties to the map, but we add anything we want to the struct.